### DISCUSSION TITLE
[Cannot Support] LLVM 12 Support

### DISCUSSION THREAD

**Discussion by ghost** [odin-lang/Odin#1038] [Proposals]
I recently got myself into a bit of a predicament: I updated all my packages on my machine without realizing that I was going to be upgrading LLVM from 11 to 12. Then, I tried to re-run make to build Odin and found that Odin doesn't yet support LLVM 12 (fair enough, it's pretty new.)
Unfortunately it seems that individually downgrading llvm to get around this temporarily is more work than it's worth, assuming LLVM 12 support comes along eventually.

Just figured I'd mark this down somewhere since I didn't see it mentioned yet. Thanks!

**Comment by gingerBill** [comment]
Unfortunately there two problems out of my hands:

Arch has forced everyone to use LLVM 12 and for some reasons you cannot get LLVM 11 anymore for that distro (what the fuck has happened here?!)
I attempted to upgrade Odin to LLVM 12 last night and it appears that LLVM 12 is fundamentally broken (with the C API) and those problems are not fixed until the next (official) version of LLVM is out (LLVM 12 is a write-off for Odin)

**Reply by ghost** [reply]
Ouch, that's worse than I was expecting :frowning: I suppose Arch had to eventually do something to tick me off :stuck_out_tongue_closed_eyes: 

As a temporary workaround I ended up building Odin in an OCI container, and copying the requisite .so files out of the container into the Odin directory, and then wrapped the Odin executable with a script that runs it with LD_LIBRARY_PATH set to load those .so's. Certainly not a pleasant solution but I guess for now it may be what people on Arch have to do.
